src/main/java/org/logstash/plugins/inputs/http/NettyHttpServer.java

```diff
@@ -0,0 +1,91 @@
+package org.logstash.plugins.inputs.http;
+
+import io.netty.bootstrap.ServerBootstrap;
+import io.netty.channel.ChannelFuture;
+import io.netty.channel.ChannelOption;
+import io.netty.channel.EventLoopGroup;
+import io.netty.channel.nio.NioEventLoopGroup;
+import io.netty.channel.socket.nio.NioServerSocketChannel;
+import org.logstash.plugins.inputs.http.util.CustomRejectedExecutionHandler;
+import org.logstash.plugins.inputs.http.util.SslSimpleBuilder;
+
+import java.io.Closeable;
+import java.util.concurrent.ArrayBlockingQueue;
+import java.util.concurrent.ThreadPoolExecutor;
+import java.util.concurrent.TimeUnit;
+
+import static org.logstash.plugins.inputs.http.util.DaemonThreadFactory.daemonThreadFactory;
+
+/**
+ * Created by joaoduarte on 11/10/2017.
+ */
+public class NettyHttpServer implements Runnable, Closeable {
+    private final ServerBootstrap serverBootstrap;
+    private final String host;
+    private final int port;
+
+    private final EventLoopGroup processorGroup;
+    private final ThreadPoolExecutor executorGroup;
+
+    public NettyHttpServer(String host, int port, IMessageHandler messageHandler,
+                           SslSimpleBuilder sslSimpleBuilder, int threads,
+                           int maxPendingRequests, int maxContentLength)
+    {
+        this.host = host;
+        this.port = port;
+        processorGroup = new NioEventLoopGroup(threads, daemonThreadFactory("http-input-processor"));
+
+        executorGroup = new ThreadPoolExecutor(threads, threads, 0, TimeUnit.MILLISECONDS,
+                new ArrayBlockingQueue<>(maxPendingRequests), daemonThreadFactory("http-input-handler-executor"),
+                new CustomRejectedExecutionHandler());
+
+        final HttpInitializer httpInitializer = new HttpInitializer(messageHandler, executorGroup);
+
+        if (sslSimpleBuilder != null) {
+            httpInitializer.enableSSL(sslSimpleBuilder);
+        }
+        httpInitializer.setMaxContentLength(maxContentLength);
+
+        serverBootstrap = new ServerBootstrap()
+                .group(processorGroup)
+                .channel(NioServerSocketChannel.class)
+                .childOption(ChannelOption.SO_KEEPALIVE, true)
```

Comment:
If we enable this, it should be able to be optional turned off.  If we hold the idle connections open (e.g. prevent the client from timing out) we could lock up the upstream system if we are blocked and the upstream system exhausts all of its connections from it's pool.

Reply:
FWIW, it looks like Netty's default (or maybe it my mac ?) is to send keep alives every 60s.  Just a mental note that client side timeout need to be greater then 60s for this to have the desired outcome.

<img width="1885" alt="image" src="https://user-images.githubusercontent.com/976291/38062106-d7343b62-32b7-11e8-8d80-2be3b9d658c4.png">